src/main/java/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,36 @@
+package service.impl;
+
+import database.StorageDealer;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import model.FruitTransaction;
+import service.ShopService;
+import strategy.OperationStrategy;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+    private final StorageDealer storageDealer;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy, StorageDealer storageDealer) {
+        this.operationStrategy = operationStrategy;
+        this.storageDealer = storageDealer;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> fruitTransactions) {
```

Comment:
The method `process()` is doing more than one thing. It's violating the Single Responsibility Principle. You're calculating the quantity of each fruit, checking the balance, and updating the database all in one method. It would be better to split these operations into separate methods.